evaluation.py

```diff
@@ -97,7 +98,40 @@ def save_stacked_evaluation_heatmaps(self, h, X, y, stacked_predict_heatmaps_fil
         img_v_resize = self._vstack_images(heatmap_imgs) 
         
         cv2.imwrite(filename, img_v_resize) 
-    
+
+    # Resources for heatmaps to keypoints
+    # https://github.com/yuanyuanli85/Stacked_Hourglass_Network_Keras/blob/eddf0ae15715a88d7859847cfff5f5092b260ae1/src/eval/heatmap_process.py#L5
+    # https://github.com/david8862/tf-keras-stacked-hourglass-keypoint-detection/blob/56707252501c73b2bf2aac8fff3e22760fd47dca/hourglass/postprocess.py#L17
+   
+    ### Returns np array of predicted keypoints from one image's heatmaps
+    def heatmaps_to_keypoints(self, heatmaps, threshold=HM_TO_KP_THRESHOLD):
+        keypoints = list()
+        for i in range(NUM_COCO_KEYPOINTS):
+            hmap = heatmaps[:,:,i]
+            # Resize heatmap from Output DIM to Input DIM
+            resized_hmap = cv2.resize(hmap, INPUT_DIM, interpolation = cv2.INTER_LINEAR)
+            # Do a heatmap blur with gaussian_filter
+            resized_hmap = gaussian_filter(resized_hmap, REVERSE_HEATMAP_SIGMA)
+
+            # Get peak point (brightest area) in heatmap with 3x3 max filter
+            peaks = self._non_max_supression(resized_hmap, threshold, windowSize=3)
+
+            # Choose the max point in heatmap (we only pick 1 keypoint in each heatmap)
+            # and get its coordinates and confidence
+            y, x = np.where(peaks == peaks.max())
+            if int(x[0]) > 0 and int(y[0]) > 0:
+                keypoints.append((int(x[0]), int(y[0]), 1))
+            else:
+                keypoints.append((0, 0, 0))
+        # Turn keypoints into np array
+        keypoints = np.array(keypoints)
```

Comment:
Are you intending for the shape to be nd.array of shape (point, xyz)?

Reply:
I just saw this now. I think `keypoints.append(tuple)` is mixing tuples and lists. Is this intentional?

Reply:
I think it should add tuples to a list and then convert to an nd array of points [x,y,z] do you think that should be handled a different way?

Reply:
I just checked and the end result is [ [x1, y1, v1], [x2, y2, v2], etc] (so it's a 2d array). I just wanted to check because the original COCO data was a 1d array of `[x1,y1,v1,x2,y2,v2,...]`.

Reply:
Oh yeah it still has to get adjusted with Rafay's undoing bounding box code which formats it like that I believe 👍 we can change it here to but it just depends on how it gets fed into that part

Reply:
Sounds good. no preference from my side

Reply:
Sounds good I'll mention it to Rafay thanks for checking!